### PR TITLE
ssh-client: fix merge order of options in run_wfunc (fix usage of mine.get in pillar)

### PR DESCRIPTION
**Pull request by ptitdoc**
### What does this PR do?
pillar.Pillar() is called with options where master_opts options are overwritten (because of dict merge/update order). As a consequence, most saltutil functions are not available in the pillar when salt-ssh client is used.

This PR change the opts merge order selectively based on specific wfunc routines (eg: mine.get, pillar.items ...)

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/36796
https://github.com/saltstack/salt/pull/41025

### Previous Behavior
master_opts are overwritten when calling mine.get or pillar.items within salt-ssh client.

master_opts takes precedence when calling mine.get or pillar.items within salt-ssh client.

### Tests written?
No

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by thatch45 — APPROVED** [review]
This is a very good solution to the issue here, thanks!

File: salt/client/ssh/__init__.py [hunk 1]
```diff
@@ -1075,8 +1075,22 @@ def run_wfunc(self):
...
+            #
+            # Depending on popts merge order, it will overwrite some options found in opts_pkg['__master_opts__']
+            master_centric_funcs = [
+              "pillar.items",
+              "mine.get"
+            ]
+
+            # Pillar compilation is a master centric operation.
+            # Master options take precedence during Pillar compilation
+            popts.update(opts_pkg['__master_opts__'])
```
Comment (isbm) [inline salt/client/ssh/__init__.py]: Q: are the **keys** of the `opts_pkg` aren't the same as in `__master_opts__`? I.e. aren't `opts_pkg` just do not override the master opts? In this case the first `update(opts_pkg)` is not really needed. But I am not entirely sure here.

**Review by isbm — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
@ptitdoc Did you see the feedback here from @isbm? Once those issues are resolved we should be able to merge this. Thanks!

File: salt/client/ssh/__init__.py [hunk 2]
```diff
@@ -1085,6 +1099,13 @@ def run_wfunc(self):
                     )
             pillar_data = pillar.compile_pillar()
 
+            # Once pillar has been compiled, restore priority of minion opts
+            if self.fun not in master_centric_funcs:
+                log.debug('%s is a minion function', self.fun)
+                popts.update(opts_pkg)
```
Comment (isbm) [inline salt/client/ssh/__init__.py]: @cachedout I am still wondering if we need that `popts.update(opts_pkg)` first time. This one is a second.

Comment (cachedout) [reply salt/client/ssh/__init__.py]: Will it do any harm though?

Comment (thatch45) [reply salt/client/ssh/__init__.py]: I think this is fine and it makes sense.

**Review by thatch45 — APPROVED** [review]

**Pull request merged**